app/web3/transactionConfirmation.ts

```diff
@@ -0,0 +1,73 @@
+/* tslint:disable:no-console */
+
+import config from "../config";
+import web3Provider from "./web3Provider";
+
+export const transactionConfirmation = async (
+  transactionHash: string,
+  transactionMinedCallback: any,
+  newBlockCallback: any
+) => {
+  console.log("waiting for transaction: " + transactionHash);
+  return new Promise((resolve, reject) => {
+    let prevBlockNo = -1;
+    let startingBlock = -1;
+    const requiredConfirmations = config.transactionSigning.numberOfConfirmations;
+    const maxNumberOfBlocksToWait = config.transactionSigning.maxNumberBlocksToWait;
+    const poll = async () => {
+      let currentBlockNo;
+      try {
+        currentBlockNo = await promisify(web3Provider.eth.getBlockNumber, []);
+      } catch (e) {
+        console.log("error in web3.eth.getBlockNumber");
+        console.log(e);
+        return reject(e);
+      }
+
+      if (startingBlock === -1) {
+        startingBlock = currentBlockNo;
+      }
+
+      if (currentBlockNo - startingBlock >= maxNumberOfBlocksToWait) {
+        return reject(
+          `Your transaction has not been mined in last ${maxNumberOfBlocksToWait} blocks`
+        );
+      }
+
+      console.log(`got block number ${currentBlockNo} prev block number ${prevBlockNo}`);
+      if (currentBlockNo !== prevBlockNo) {
+        prevBlockNo = currentBlockNo;
+        newBlockCallback(currentBlockNo);
+
+        try {
+          const transaction = await promisify(web3Provider.eth.getTransaction, [transactionHash]);
+          console.log(`got transaction with block number: ${transaction.blockNumber}`);
+          if (transaction.blockNumber != null) {
+            transactionMinedCallback(transaction.blockNumber);
+            if (currentBlockNo - transaction.blockNumber >= requiredConfirmations) {
+              console.log("we have enough confirmations we can move on");
+              return resolve();
+            }
+          }
+        } catch (e) {
+          console.log("error in web3.eth.getTransaction");
+          console.log(e);
+          return reject(e);
+        }
+      }
+      window.setTimeout(poll, 1000);
```

Comment:
what about making this timeout bigger and using `setInterval`. Average time between blocks in ethereum is 9 seconds i belive (you can double check it ;) )

Reply:
As sometimes those calls take some time we didn't want them to call every second but second after last call ended. That was reason behind using `setTimeout`

Reply:
That's why I thought that we should make it way more than 1 second like 8? and these calls should be fast like few hundred ms max. So you can neglect it. What do you think?